### ISSUE TITLE
Can clients also attach a profile to entity bodies?

### ISSUE THREAD

**Issue by RubenVerborgh** [ProfileNegotiation/I-D-Accept--Schema#15]
If a client performs a POST request, should it be able to add a profile to it in addition to a media type?

**Comment by larsgsvensson** [comment]
Yes, I'd say that should be possible. I think that's something missing in LDP...

**Comment by ericprud** [comment]
Can you wedge profile parameters into the POST's `Content-Type: text/turtle; profile=foo; profile=bar`?
If you want to control the POST response representation, I think you want to define a new `Prefer: profile=` parameter to extend the parameters defined in [RFC7240](https://tools.ietf.org/html/rfc7240).

Headers currently used by LDP:
* `Link: ldp:BasicContainer; rel="type"` - (response) advertise interface, (request) require (e.g. resource or container) behavior.
* `Accept-Patch: application/patch-format-1, text/patch-format-2` - (response)
* `Slug: newFileName` - (POST request) advise where to create resource ([RFC7023](https://tools.ietf.org/html/rfc5023#section-9.7))
* `Location: IRI` - (201 response) - identify location of 201 created resource.
* `Link: IRI; rel="describedBy"` - (201 response) - identify location of a 201 created LDP resource corresponding to an accepted `ldp:NonRDFSource`.
* `Accept-post: text/turtle, application/ld+json` (replaces [expired draft](https://datatracker.ietf.org/doc/draft-wilde-accept-post/)) - (response) tells clients what to POST to a service.
* `Prefer: return=representation; include=ldp:PreferMinimalContainer` - (request) advises server of preferred "behavior". LDP [adds a three values](https://www.w3.org/TR/ldp/#h-prefer-uris) for `Prefer: return=representation; {include,omit}=` for selecting subsets of graphs: `ldp:PreferContainment`, `ldp:PreferMembership`, `ldp:PreferMinimalContainer`
* `Preference-Applied: return=representation` - (response)
* `ETag: "something-opaque"` (response) and `If-Match: "something-opaque"` (request)

**Comment by larsgsvensson** [comment]
> Can you wedge profile parameters into the POST's Content-Type: text/turtle; profile=foo; profile=bar

I don't think that works for all media types. As with `Accept`, `Content-Type` only accepts the `profile` parameter if it's allowed in the media type registration.

Regarding what is returned after POST: How does a client usually control that? Is it possible to send an `Accept`-Header with a POST request in order to say which media type I want to get back?

**Comment by ericprud** [comment]
Yes. [RFC2616](https://tools.ietf.org/html/rfc2616) (now as [RFC7231](https://tools.ietf.org/html/rfc7231) separates [method behaviors](https://tools.ietf.org/html/rfc7231#section-4.3.3) from [header semantics](https://tools.ietf.org/html/rfc7231#section-5.3.2) with the intent that all sensible combinations be defined. The LDP spec has [examples of this](https://www.w3.org/TR/ldp/#ldpc-ex-membership-add-new-liability-req).

As a practical point, while Accept headers can include parameters and some media type registrations include a `profile` parameter, I've never seen a mention of using a profile in an Accept header. I believe we're forging new ground here and will need to provide useful anchor text for future media type registrations to reference if they expect profile to be used to switch between iso-semantic forms.
